src/Chapter2.hs

```diff
@@ -620,7 +640,15 @@ Write a function that takes elements of a list only on even positions.
 >>> takeEven [2, 1, 3, 5, 4]
 [2,3,4]
 -}
-takeEven = error "takeEven: Not implemented!"
+takeEven :: [Int] -> [Int]
+takeEven [] = []
+takeEven l = go 0 [] l
+  where
+    go :: Int -> [Int] -> [Int] -> [Int]
+    go index acc l
+      | null l = acc
+      | index `mod` 2 == 0 = go (index + 1) (acc ++ (subList 0 0 l)) (tail l)
```

Comment:
These are really costly operations: `++`, `acc`, `sublist`, `tail`.

The most efficient solution here would be to use pattern matching. You can represent even element with pattern matching! For that you would need just 3 cases 🙂 Could you guess how now?